zerver/views/typing.py

```diff
@@ -15,6 +16,23 @@ def send_notification_backend(
         request: HttpRequest,
         user_profile: UserProfile,
         operator: str=REQ('op', str_validator=check_string_in(VALID_OPERATOR_TYPES)),
-        notification_to: List[int]=REQ('to', validator=check_list(check_int))) -> HttpResponse:
-    check_send_typing_notification(user_profile, notification_to, operator)
+        notification_to: Optional[List[int]]=REQ('to', validator=check_list(check_int), default=None),
+        to_stream: Optional[int]=REQ('stream_id', validator=check_int, default=None),
+        to_topic: Optional[str]=REQ('topic', default=None)) -> HttpResponse:
+
+    bad_arguments_error = _("Bad arguments. Should have 'to' or both 'stream_id' and 'topic'.")
+
+    if not any([notification_to, to_stream, to_topic]):
+        return json_error(bad_arguments_error)
+
+    if all([notification_to, to_stream, to_topic]):
+        return json_error(bad_arguments_error)
+
+    if notification_to and not (to_stream or to_topic):
+        check_send_typing_notification(user_profile, notification_to, operator)
+    elif to_stream and to_topic:
+        check_send_stream_typing_notification(user_profile, operator, to_stream, to_topic)
+    else:
+        return json_error(bad_arguments_error)
```

Comment:
Perhaps it would  be cleaner to write this somewhat like this?
```
    if any(
        notification_to and (to_stream or to_topic),
        not notification_to and not(to_stream and to_topic)
    ):
        return json_error(bad_arguments_error)

    if notification_to:
        check_send_typing_notification(user_profile, notification_to, operator)
    else:
        check_send_stream_typing_notification(user_profile, operator, to_stream, to_topic)
```